src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +630,560 @@ def test_call_get_callees_local(fortran_reader):
     assert result == [psyir.walk(Routine)[1]]
 
 
+def test_call_get_callee_1_simple_match(fortran_reader):
+    '''
+    Check that the right routine has been found for a single routine
+    implementation.
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(e, f, g)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b, c
+  end subroutine
+
+end module some_mod'''
+
+    psyir = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = psyir.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+
+    (result, _) = call_foo.get_callee()
+
+    routine_match: Routine = psyir.walk(Routine)[1]
+    assert result is routine_match
+
+
+def test_call_get_callee_2_optional_args(fortran_reader):
+    '''
+    Check that optional arguments have been correlated correctly.
+    '''
+    code = '''
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    call foo(e, f)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b
+    integer, optional :: c
+  end subroutine
+
+end module some_mod'''
+
+    root_node: Node = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = root_node.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    routine_match: Routine = root_node.walk(Routine)[1]
+    assert routine_match.name == "foo"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+    assert call_foo.routine.name == "foo"
+
+    (result, arg_idx_list) = call_foo.get_callee()
+    result: Routine
+
+    assert len(arg_idx_list) == 2
+    assert arg_idx_list[0] == 0
+    assert arg_idx_list[1] == 1
+
+    assert result is routine_match
+
+
+def test_call_get_callee_3_trigger_error(fortran_reader):
+    '''
+    Test which is supposed to trigger an error.
```

Comment:
Please specify the error - e.g. "error when no matching routine is found" or something?

Reply:
OK.